front/src/applications/editor/tools/rangeEdition/components/RangeEditionTrackRangeList.tsx

```diff
@@ -30,48 +33,43 @@ const TrackRangesList = () => {
   const ranges = entity.properties.track_ranges || [];
   const displayedRanges = showAll ? ranges : ranges.slice(0, DEFAULT_DISPLAYED_RANGES_COUNT);
 
-  const rangesByRoute = groupBy(displayedRanges, (range) => {
-    const isCurrentRange = compareTrackRange(range);
-    // eslint-disable-next-line no-restricted-syntax
-    for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
-      if (isCurrentRange(routeExtra[routeKey]) || trackRanges.some(isCurrentRange)) {
-        return routeKey;
-      }
+  function makeFullTrackList() {
+    if (withRouteName) {
```

Comment:
Why use withRouteName here and not directly speedRestrictionTool ?
```suggestion
    if (speedRestrictionTool) {
```

Reply:
ok it’s changed